lib/RestructureCFG/InlineDivergentScopesPass.cpp

```diff
@@ -366,31 +507,28 @@ static bool tryIDS(Function &F, BasicBlock *PlaceHolderTarget) {
       std::optional<DivergenceDescriptor>
         DivergenceDescriptor = electDivergence(Candidate, Exit, DomTree);
 
-      // If we have found a divergence for the exit under analysis, we proceed
-      // to perform IDS
+      // If we have found a divergence for the exit under analysis, we add it to
+      // the `DivergenceDescriptors`
       if (DivergenceDescriptor) {
-        performIDS(*DivergenceDescriptor, PlaceHolderTarget);
-
-        // Empirically, we have found that restarting the analysis, by
-        // recollecting the exit nodes in the `ScopeGraph`, is faster than
-        // continuing with the processing of all the exits already collected.
-        // Therefore, we should not really try to optimize this, unless we
-        // find new evidence that this is better.
-        return true;
+        DivergenceDescriptors.push_back(*DivergenceDescriptor);
+
+        // We have found a `Divergence` for the current `Exit`, so we can move
+        // to the next one
+        break;
       }
     }
   }
 
-  // If no change has been performed, we return `false` to signal this fact
-  return false;
+  return tryMultipleIDS(SGBuilder, DivergenceDescriptors, PlaceHolderTarget);
 }
 
 /// Implementation class used to run the `IDS` transformation
 class InlineDivergentScopesImpl {
   Function &F;
+  ScopeGraphBuilder SGBuilder;
```

Comment:
I think this can be `const` too.

Reply:
This member is now `const` here https://github.com/revng/revng/blob/3a9379ef740d1d2b9f7bcad02cc962719251e02a/lib/RestructureCFG/InlineDivergentScopesPass.cpp#L597